csharp-basics/exercises/Scooters/Scooters/ScooterService.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections.Generic;
+using System.Collections;
+using System.Linq;
+using System;
+
+namespace Scooters
+{
+    public class ScooterService : IScooterService
+    {
+        private static List<Scooter> _scooterList = new List<Scooter>();
+
+        public ScooterService()
+        {
+        }
+
+        public void AddScooter(string id, decimal pricePerMinute)
+        {
+            if (!ScooterIdExists(id))
+            {
+                _scooterList.Add(new Scooter(id, pricePerMinute));
+            }
+            else throw new DuplicateScooterIdException("Scooter id exists!");
+        }
+
+        public bool ScooterIdExists(string scooterId)
+        {
+            return _scooterList.Find(x => x.Id == scooterId)==null?false:true;
+
+            /*bool result = false;
+            foreach (Scooter scooter in _scooterList)
+            {
+                if (scooter.Id == id) result = true;
+            }
+
+            return result;*/
+        }
+
+        public void RemoveScooter(string id)
+        {
+            Scooter scooter = GetScooterById(id);
+            if (scooter.IsActive)
+            {
+                if (!scooter.IsRented)
+                {
+                    scooter.IsActive = false;
+                }
+                else throw new ScooterRentedException("Scooter is rented!");
+            }
+            else throw new ScooterActivityException("Scooter is not active!");
+        }
+
+        public void ReactivateScooter(string id)
```

Comment:
kāda doma šai metodei? 
Tev jau ir IsRented parametrs kas manuprāt darbojas tāpat.

Reply:
Šai metodei tieši arī ir jega. Ja skuteris tika izdzēsts: "!IsActive". Piemēram, vēlāk tas tiek salabots, un var tikt atpakaļ sarakstā ar ReactivateScooter(id). Tad šo id var atkal izmantot for Rental. Šī ir jauna metode, kas domāta optimizācijai, kura, pēc manām domām, ir loģiska, bet nav iespējama ar uzdevumā dotajām metodēm.